skbio/io/format/genbank.py

```diff
@@ -609,77 +610,86 @@ def _parse_features(lines, length):
+def _parse_single_feature(lines, imd=None):
     '''Parse a feature.
 
-    Returns
-    -------
-    tuple
-        Tuple of a dict of `metadata` and a pandas.Series of
-        `positional_metadata` for the feature.
+    Parse a feature and add it to ``IntervalMetadata`` object.
 
+    Parameters
+    ----------
+    imd : IntervalMetadata
     '''
-    feature = {}
-    feature['index_'] = index
     # each component of a feature starts with '/', except the 1st
     # component of location.
     section_splitter = _yield_section(
         lambda x: x.startswith('/'), strip=True)
-    first = True
-    for section in section_splitter(lines):
-        if first:
-            # first section is the Location string
-            first = False
-            type, location = _parse_section_default(
-                section, join_delimitor='', return_label=True)
-            feature['type_'] = type
-            feature['location'] = location
-            loc, loc_pmd = _parse_loc_str(location, length)
-            feature.update(loc)
+    section_iter = section_splitter(lines)
+
+    # 1st section is location
+    section = next(section_iter)
+    feature_type, feature_loc = _parse_section_default(
+        section, join_delimitor='', return_label=True)
+
+    metadata = {'__type__': feature_type, '__location__': feature_loc}
+
+    if imd is not None:
+        intvl = imd.add(*_parse_loc_str(feature_loc))
+
+    for section in section_iter:
+        # following sections are Qualifiers
+        k, v = _parse_section_default(
+            section, label_delimitor='=',
+            join_delimitor=' ', return_label=True)
+        # 1st char is '/'
+        k = k[1:]
+
+        # some Qualifiers can appear multiple times
+        if k in metadata:
+            if not isinstance(metadata[k], list):
+                metadata[k] = [metadata[k]]
+            metadata[k].append(v)
         else:
-            # following sections are Qualifiers
-            k, v = _parse_section_default(
-                section, label_delimitor='=',
-                join_delimitor=' ', return_label=True)
-            k = k[1:]
-
-            # some Qualifiers can appear multiple times
-            if k in feature:
-                if not isinstance(feature[k], list):
-                    feature[k] = [feature[k]]
-                feature[k].append(v)
-            else:
-                feature[k] = v
-    return feature, loc_pmd
+            metadata[k] = v
+
+    if imd is None:
+        return metadata
+    else:
+        intvl.metadata.update(metadata)
 
 
-def _serialize_single_feature(obj, indent=21):
+def _serialize_single_feature(intvl_md, indent=21):
+    '''
+    Parameters
+    ----------
+    intvl_md : dict
+        Interval.metadata
+    '''
     padding = ' ' * 8
     qualifiers = []
-    for k in sorted(obj):
-        if k.endswith('_') or k in ('location', 'type'):
+    for k in sorted(intvl_md):
+        if k.endswith('__') and k.startswith('__'):
```

Comment:
Since we are explicitly searching for keywords with `__` in front, shouldn't we add a note in the docs about this?  The malevolent user may try using keywords with `__`.  

If we are extremely paranoid about this, maybe we should even raise an error that words that being and end with `__` are reserved?